### PR TITLE
Admin upgrade downgrade users #56

### LINKED ISSUE #53: User Story 60, Admin is redirected from Merchant Dashboard to User profile
```
As an admin user
If I visit a merchant dashboard, but that merchant is a regular user
Then I am redirected to the appropriate user profile page.

eg, if I visit "/admin/merchants/7" but that merchant is a regular user
then I am redirected to "/admin/users/7" and see their user profile page
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #59

child of #125

### LINKED ISSUE #54: User Story 59, Admin can downgrade a merchant to regular user
```
As an admin user
When I visit a merchant's dashboard ("/admin/merchants/6")
I see a link to "downgrade" the merchant's account to become a regular user
The merchant themselves do NOT see this link
When I click on that link
I am redirected to ("/admin/users/6") because the merchant is now a regular user
And I see a flash message indicating the user has been downgraded
The next time this user logs in they are no longer a merchant
All items for sale by this user are disabled
Only admins can see the "downgrade" button
Only admins can reach any route necessary to downgrade the merchant to user status
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #59

child of #125

### LINKED ISSUE #55: User Story 58, Admin is redirected from User profile to Merchant dashboard
```
As an admin user
If I visit a profile page for a user, but that user is a merchant
Then I am redirected to the appropriate merchant dashboard page.
eg, if I visit "/admin/users/7" but that user is a merchant
Then I am redirected to "/admin/merchants/7"
And I see their merchant dashboard page
```

DO THESE STORIES FIRST:
- User Story 56, Admin User Profile Page
- User Story 38, Admin can see a merchant's dashboard

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #59

child of #125

depends on #57

depends on #79

### LINKED ISSUE #56: User Story 57, Admin can make a User a Merchant
```
As an admin user
When I visit a user's profile page ("/admin/users/5")
I see a link to "upgrade" the user's account to become a merchant
When I click on that link
I am redirected to ("/admin/merchants/5") because the user is now a merchant
And I see a flash message indicating the user has been upgraded
The next time this user logs in they are now a merchant
Only admins can reach any route necessary to upgrade the user to merchant status
```

DO THESE STORIES FIRST:
- User Story 55, Admin User Index Page
- User Story 38, Admin can see a merchant's dashboard

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #59

child of #125

depends on #58

depends on #79

### PR DESCRIPTION
**Pull request by Vjp888**
Adds the ability for an admin to upgrade and downgrade users from merchant to user and vice versa.

Also adds the ability for the admin to be redirected if they enter in the incorrect URI.
If they enter '/admin/users/:id' but the ID is of a merchant they are redirected to the admin merchant path. This is also the case for a user in the merchants URI

All tests passing

closes #56 
closes #55 
closes #54 
closes #53

### CODE REVIEW

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -13,4 +16,19 @@ def update
     merchant.save
     redirect_to admin_merchants_path
   end
+
+  def downgrade
```
Comment (wipegup) [inline app/controllers/admin/merchants_controller.rb:20]: This all looks good, however, for a re-factor we might make `downgrade` and `upgrade` into a more RESTful `update`.  

Nothing that should block a merge though

**Review by wipegup — APPROVED** [review]

**Review by whois-jon-peterson — APPROVED** [review]
Agreed with @wipegup remarks on `update` vs `downgrade`/`upgrade`

**Review by cnodland — APPROVED** [review]
